hypervisor/dm/vpic.c

```diff
@@ -51,10 +51,10 @@ enum irqstate {
 /*
  * Loop over all the pins in priority order from highest to lowest.
  */
-#define	PIC_PIN_FOREACH(pinvar, i8259, tmpvar)			\
-	for (tmpvar = 0U, pinvar = (i8259->lowprio + 1U) & 0x7U;	\
-	    tmpvar < NR_VPIC_PINS_PER_CHIP;				\
-	    tmpvar++, pinvar = (pinvar + 1U) & 0x7U)
+#define	PIC_PIN_FOREACH(pinvar, i8259, tmpvar)				\
+	for ((tmpvar) = 0U, (pinvar) = ((i8259)->lowprio + 1U) & 0x7U;	\
+	    (tmpvar) < NR_VPIC_PINS_PER_CHIP;				\
+	    (tmpvar)++, (pinvar) = ((pinvar) + 1U) & 0x7U)
```

Comment:
This MACRO has other issues:   1) 2 operations in  for ( X, Y; ..; M, N).  I think we also need to solve this issue

Reply:
A 'foreach' macro for array/list enumeration compacts multiple expressions needed into one for-loop header, inevitably introducing violations with regard to the for-loop well-formedness. 

However, enforcing well-formed for loop in such case means the introduction of duplicated code to every place using these 'foreach' helpers, which can hurt readability. As we have discussed earlier, we are considering deviating the well-formed rule, but only for these 'foreach' cases.